metadata/metadata.go

```diff
@@ -1727,6 +1827,270 @@ func (serv *MetadataServer) GetEntities(stream pb.Metadata_GetEntitiesServer) er
+	if err != nil {
+		return nil, fmt.Errorf("trigger does not exist: %v", err)
+	}
+
+	// Get the resource
+	resourceID := ResourceID{Name: tr.Resource.Resource.Name, Variant: tr.Resource.Resource.Variant, Type: ResourceType(tr.Resource.ResourceType)}
+	resourceRecord, err := serv.lookup.Lookup(resourceID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Remove the resource from the trigger
+	trigger_taskIDs := triggerRecord.(*triggerResource).serialized.TaskIds
+	trigger_jobIDs := triggerRecord.(*triggerResource).serialized.JobIds
+
+	// Remove the trigger from the resource
+	switch r := resourceRecord.(type) {
+	case *featureVariantResource:
+		r.serialized.Trigger, err = removeStringFromList(r.serialized.Trigger, tr.Trigger.Name)
+		if err != nil {
+			return nil, err
+		}
+		triggerRecord.(*triggerResource).serialized.TaskIds, err = removeIntFromList(trigger_taskIDs, r.serialized.TaskId)
+		if err != nil {
+			return nil, err
+		}
+		triggerRecord.(*triggerResource).serialized.JobIds, err = removeIntFromList(trigger_jobIDs, r.serialized.JobId)
+		if err != nil {
+			return nil, err
+		}
+	case *trainingSetVariantResource:
+		r.serialized.Trigger, err = removeStringFromList(r.serialized.Trigger, tr.Trigger.Name)
+		if err != nil {
+			return nil, err
+		}
+		triggerRecord.(*triggerResource).serialized.TaskIds, err = removeIntFromList(trigger_taskIDs, r.serialized.TaskId)
+		if err != nil {
+			return nil, err
+		}
+		triggerRecord.(*triggerResource).serialized.JobIds, err = removeIntFromList(trigger_jobIDs, r.serialized.JobId)
+		if err != nil {
+			return nil, err
+		}
+	case *sourceVariantResource:
+		r.serialized.Trigger, err = removeStringFromList(r.serialized.Trigger, tr.Trigger.Name)
+		if err != nil {
+			return nil, err
+		}
+		triggerRecord.(*triggerResource).serialized.TaskIds, err = removeIntFromList(trigger_taskIDs, r.serialized.TaskId)
+		if err != nil {
+			return nil, err
+		}
+		triggerRecord.(*triggerResource).serialized.JobIds, err = removeIntFromList(trigger_jobIDs, r.serialized.JobId)
+		if err != nil {
+			return nil, err
+		}
+	case *labelVariantResource:
+		r.serialized.Trigger, err = removeStringFromList(r.serialized.Trigger, tr.Trigger.Name)
+		if err != nil {
+			return nil, err
+		}
+		triggerRecord.(*triggerResource).serialized.TaskIds, err = removeIntFromList(trigger_taskIDs, r.serialized.TaskId)
+		if err != nil {
+			return nil, err
+		}
+		triggerRecord.(*triggerResource).serialized.JobIds, err = removeIntFromList(trigger_jobIDs, r.serialized.JobId)
+		if err != nil {
+			return nil, err
+		}
+	default:
+		return nil, fmt.Errorf("could not assert resource")
+	}
+
+	// Save the resource back
+	err = serv.lookup.Set(resourceID, resourceRecord)
+	if err != nil {
+		return nil, err
+	}
+	// Save the trigger back
+	err = serv.lookup.Set(triggerID, triggerRecord)
+	if err != nil {
+		return nil, err
+	}
+	return &pb.Empty{}, nil
+}
+
+func removeIntFromList(slice []int32, id int32) ([]int32, error) {
```

Comment:
These should probably in the helpers directory and can use generics instead of having one for each type

Reply:
+1 on this